app/controllers/spaces_controller.rb

```diff
@@ -17,20 +17,19 @@ def update
   end
 
   def create
-    @space = Space.new(space_params)
     skip_policy_scope
-    authorize(@space)
-    Blueprint.new(client:
-      { name: space_params[:name],
-        space: SystemTestSpace::DEFAULT_SPACE_CONFIG.merge(space_params).with_indifferent_access }).find_or_create!
+    client = Client.find_or_create_by(name: space_params[:name])
+    authorize(client)
```

Comment:
FWIW, `authorize` returns the thing that it authorizes, so you can use it for chaining and assigns, so you could do:
```ruby
client = authorize(Client.find_or_create_by(name: space_params[:name]))
```